app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -94,6 +94,7 @@ public Retrofit getRetrofit() {
                 HttpLoggingInterceptor.Level.NONE);
         final OkHttpClient okHttpClient = new OkHttpClient.Builder()
                 .addInterceptor(logging)
+                .readTimeout(20, TimeUnit.SECONDS)
```

Comment:
can you do a negative test for this somehow? To see what actually happens after let's say 15 seconds. I just want to get sure, that the app doesn't crash (e.g. ANR)

Reply:
@tuancoltech What is the default timeout value by the way? 5 seconds?

Reply:
@sladomic The negative test results in a `java.net.SocketTimeoutException` being thrown. The result is not an app crash, but just a blank page being displayed. Instead, we should probably display an error message, and redirect the user back to the original state (so that she can try the request again).

Reply:
@sladomic Basically every Retrofit request is executed on background thread, so there is no need to worry about ANR when adjusting read timeout.
@jo-elimuai The default connect/read/write timeout in Retrofit is 10 seconds.
Kindly refer to this document: https://futurestud.io/tutorials/retrofit-2-customize-network-timeouts